apps/mobile/src/screens/ProfilePicturePickerScreen/ProfilePicturePickerGrid.tsx

```diff
@@ -83,6 +85,11 @@ export function ProfilePicturePickerGrid({
 
   const filteredTokens = useMemo(() => {
     return tokens
+      .filter((token) => {
+        const isSpam = token.isSpamByProvider || token.isSpamByUser;
+
+        return !isSpam;
+      })
```

Comment:
nit: many `.filter` passes can hurt performance on large arrays

prefer to have all criteria included as part of a single filter pass